react-common/components/controls/FocusTrap.tsx

```diff
@@ -18,13 +19,26 @@ export const FocusTrap = (props: FocusTrapProps) => {
         onEscape,
         arrowKeyNavigation,
         dontStealFocus,
-        includeOutsideTabOrder
+        includeOutsideTabOrder,
+        dontRestoreFocus
     } = props;
 
     let container: HTMLDivElement;
-
+    const previouslyFocused = React.useRef<Element>(document.activeElement);
     const [stoleFocus, setStoleFocus] = React.useState(false);
 
+    React.useEffect(() => {
+        return () => {
+            if (!dontRestoreFocus) {
+                let toFocus = previouslyFocused.current as HTMLElement
+                while (!isFocusable(toFocus)) {
+                    toFocus = toFocus.parentElement;
+                }
+                toFocus.focus()
+            }
+        }
+    }, [previouslyFocused])
```

Comment:
you should remove this from the dependency array. also, probably need to make sure that previouslyFocused.current is actually defined.